Assets/i5 Toolkit for Unity/Runtime/OpenID Connect/Scripts/OIDC Providers/Google/GoogleOIDCProvider.cs

```diff
@@ -0,0 +1,137 @@
+﻿using i5.Toolkit.Core.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+using System.Security.Cryptography;
+
+namespace i5.Toolkit.Core.OpenIDConnectClient
+{
+    /// <summary>
+    /// Implementation of the OpenID Connect Learning Layers Provider
+    /// More information can be found here: https://auth.las2peer.org/auth/
+    /// </summary>
+    public class GoogleOidcProvider : AbstractOidcProvider
+    {
+        /// <summary>
+        /// The generated valid session token
+        /// </summary>
+        private string state;
+
+        /// <summary>
+        /// Creates a new instance of the Learning Layers client
+        /// </summary>
+        public GoogleOidcProvider() : base()
+        {
+            serverName = "https://accounts.google.com";
+        }
+
+        /// <summary>
+        /// Creates a sequence of 30 random numbers which is used as the session token
+        /// </summary>
+        public void GenerateCSRFToken()
+        {
+            RNGCryptoServiceProvider rng = new RNGCryptoServiceProvider();
+            byte[] randomNumber = new byte[30];
+            rng.GetBytes(randomNumber);
+            string token = "";
+            for(int i=0;i<30;i++)
+            {
+                token += (randomNumber[i] % 10).ToString();
+            }
+            state = token;
```

Comment:
I think this only works on Windows systems because of the RNGCryptoServiceProvider, right?

Maybe we should put an alternative method in place for other platforms, using Unity's [conditional compiling](https://docs.unity3d.com/Manual/PlatformDependentCompilation.html). This way, we could differentiate Windows platforms from other platforms and it would work on all of them. Granted, it would likely not be as secure since we would probably have to use the normal random function for the other platforms.

Reply:
We noticed that the RNGCryptoServiceProvider is obsolete and replaced it with the equivalent RandomNumberGenerator from `System.Security.Cryptography`. Additionally, as suggested, this method is now only used in the Unity Editor and for the Windows Standalone Platform, for all other platforms the generic random number generator of Unity Engine is used. It might be, that RandomNumberGenerator works on Mac and iOS too, as it can be used with Xamarin.iOS and Xamarin.Mac. However, we could not test this and decided to not include it for now.